app/openvpn_client_lwt.ml

```diff
@@ -88,17 +110,41 @@ let read_from_fd fd =
       let cs = Cstruct.of_bytes ~len:count buf in
       Logs.debug (fun m -> m "read %d bytes" count) ;
       Lwt.return cs)
-  |> Lwt_result.map_err (fun e ->
-      Rresult.R.msgf "read error %s" (Printexc.to_string e))
+  |> Lwt_result.map_err (fun e -> `Msg (Printexc.to_string e))
 
-let rec reader mvar fd =
+let rec reader_tcp mvar fd =
   read_from_fd fd >>= function
   | Error `Msg msg ->
     Logs.err (fun m -> m "read error from remote %s" msg);
     Lwt_mvar.put mvar `Connection_failed
   | Ok data ->
     Lwt_mvar.put mvar (`Data data) >>= fun () ->
-    reader mvar fd
+    reader_tcp mvar fd
+
+let read_udp (sa, fd) =
+  Lwt_result.catch (
+    let bufsize = 2048 in
```

Comment:
since we don't in general know the MTU, i opted for a single static bytes.t (of size 65535), since the data passed out is a copy (Cstruct.of_bytes) anyways.